doc/asic-fw-update/asic-fw-update-config.md

```diff
@@ -0,0 +1,285 @@
+*/host/mlnx/asic*. This location will survive system upgrade, so system will be
+able to update ASIC FW even after system upgrade. See the new flow here:
+[Figure 2. ASIC firmware update controlled flow
+](#figure-2-asic-firmware-update-controlled-flow).
+
+<!-- omit in toc -->
+###### Figure 2. ASIC firmware update controlled flow
+<p align=center>
+  <img src="img/asic-firmware-update-new-sequence.png"
+       alt="Figure 2. ASIC firmware update controlled flow">
+</p>
+
+### Configuration and management
+#### CLI Enhancements
+This feature does not bring up any new CLI commands. But based on the caurrent
+SONiC CLI tree we can propose the next command tree.
+
+<!-- omit in toc -->
+#### CLI
+Proposed ASIC firmware control CLI tree:
+```
+config
+\-- firmware
+    \-- asic
+        |-- auto-apdate [ enable | disable ]
+        \-- default [ image | user ]
+```
+
+CLI helptext:
+```
+admin@sonic:~$ sudo config firmware
+Usage: config firmware asic [OPTIONS] COMMAND [ARGS]...
+
+  AAA command line
+
+Options:
+  -h, -?, --help  Show this message and exit.
+
+Commands:
+  auto-update     Control ASIC firmware auto update [ enable | disable ]
+  default         Set default firmware source [ image | user ]
+```
+
+#### Config DB Enhancements
+To be able to configure ASIC firmware update flow we need a new ConfigDB table.
+Basically, the table is bery simple it will have such structure:
+* CONFIG_DB
+  * FIRMWARE
+    * asic
+      * auto_update
+      * default 
+
+Here is a YANG model of a new table:
+```
+module sonic-firmware {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-firmware";
+    prefix firmware;
+
+    description "FIRMWARE YANG Module for SONiC-based OS";
+
+    revision 2022-05-05 {
+        description "First Revision";
+    }
+
+    typedef auto-update {
+        description "Configuration to set if firmware is going to be updated
+                     automatically";
+        type string {
+            pattern "enable|disable";
+        }
+    }
+
+    typedef default-source {
+        description "Configuration to set firmware source: system image or 
+                     user-defined";
+        type string {
+            pattern "image|user";
+        }
+    }
+
+    container sonic-firmware {
+
+        container FIRMWARE {
+
+            description "FIRMWARE part of config_db.json";
+
+            container asic {
+
+                leaf auto_update {
+                    type auto-update;
+                    default "enable";
+                }
+
+                leaf default {
+                    type default-source;
+                    default "image";
+                }
+
+            }
+
+        }
+        /* end of container FIRMWARE */
+    }
+    /* end of top level container */
+}
+/* end of module nvue */
+```
+
+#### Feature configuration
+<!-- omit in toc -->
+##### Disable FW auto-update
+```
+redis-cli -n 4 HSET "FIRMWARE|asic" "auto_update" "disable"
```

Comment:
Please don't give examples using redis-cli. The redis-cli is not meant to be an interface for the user. Give examples using the config commands.